Fix removed-in-Python-3.12 deprecations

gsnedders opened the pull request:
#### 93f8745ec911a80187109e33a70172e08fdd6583
<pre>
Fix removed-in-Python-3.12 deprecations
<a href="https://bugs.webkit.org/show_bug.cgi?id=260878">https://bugs.webkit.org/show_bug.cgi?id=260878</a>

Reviewed by Michael Catanzaro.

This largely just deals with the long-deprecated unittest deprecations:

 * assertRegexpMatches -&gt; assertRegex
 * assertNotRegexpMatches -&gt; assertNotRegex
 * assertRaisesRegexp -&gt; assertRaisesRegex
 * assertEquals -&gt; assertEqual

Additionally, this gets rid of usage of distutils (gone in Python
3.12) and imp (also gone in Python 3.12).

* Source/WebKit/Scripts/webkit/parser_unittest.py:
(UnsupportedPrecompilerDirectiveTest.assertRaisesRegex):
(UnsupportedPrecompilerDirectiveTest.test_error_at_else):
(UnsupportedPrecompilerDirectiveTest.test_error_at_elif):
* Tools/Scripts/webkitpy/benchmark_runner/benchmark_results_unittest.py:
(BenchmarkResultsTest.assertRaisesRegex):
(BenchmarkResultsTest.test_init):
* Tools/Scripts/webkitpy/benchmark_runner/http_server_driver/__init__.py:
* Tools/Scripts/webkitpy/benchmark_runner/utils.py:
(load_source):
(load_subclasses):
* Tools/Scripts/webkitpy/common/checkout/scm/scm_unittest.py:
(SCMTest.assertRegex):
(SCMTest):
(SCMTest.assertNotRegex):
(SCMTest._shared_test_diff_for_revision):
(test_commit_logs):
(GitTest.test_create_patch):
(GitTest.test_create_patch_with_git_index):
(GitTest.test_rename_files):
(GitSVNTest.test_delete_branch):
(GitSVNTest.test_remote_merge_base):
(GitSVNTest.test_commit_with_message_working_copy_only):
(GitSVNTest.test_commit_with_message):
(GitSVNTest.test_commit_with_message_git_commit):
(GitSVNTest.test_commit_with_message_git_commit_range):
(GitSVNTest.test_commit_with_message_only_local_commit):
(GitSVNTest.test_commit_with_message_multiple_local_commits_and_working_copy):
(GitSVNTest.test_commit_with_message_multiple_local_commits_always_squash):
(GitSVNTest.test_commit_with_message_multiple_local_commits):
(GitSVNTest.test_commit_with_message_not_synced):
(GitSVNTest.test_create_patch_local_plus_working_copy):
(GitSVNTest.test_create_patch):
(GitSVNTest.test_create_patch_after_merge):
(GitSVNTest.test_create_patch_with_changed_files):
(GitSVNTest.test_create_patch_git_commit):
(GitSVNTest.test_create_patch_git_commit_range):
(GitSVNTest.test_create_patch_working_copy_only):
(GitSVNTest.test_create_patch_multiple_local_commits):
(GitSVNTest.test_create_patch_not_synced):
(GitSVNTest.test_create_binary_patch):
* Tools/Scripts/webkitpy/common/prettypatch_unittest.py:
(PrettyPatchTest.check_ruby):
* Tools/Scripts/webkitpy/common/watchlist/watchlistparser_unittest.py:
(WatchListParserTest):
(WatchListParserTest.setUp):
* Tools/Scripts/webkitpy/common/webkitunittest.py: Removed.
* Tools/Scripts/webkitpy/layout_tests/servers/web_platform_test_server_unittest.py:
* Tools/Scripts/webkitpy/port/xvfbdriver_unittest.py:
(XvfbDriverTest.assertRaisesRegex):
(XvfbDriverTest.test_xvfb_not_replying):
* Tools/Scripts/webkitpy/tool/commands/applywatchlistlocal_unittest.py:
(ApplyWatchListLocalTest.assertRaisesRegex):
(test_args_parsing_with_two_bugs):
* Tools/Scripts/webkitpy/tool/commands/commandtest.py:
(CommandsTest):
* Tools/Scripts/webkitpy/w3c/test_converter_unittest.py:
(verify_test_harness_paths):

Canonical link: <a href="https://commits.webkit.org/267496@main">https://commits.webkit.org/267496@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ee2b1c027b6048099e29b81b6e726aea69c46fc5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16640 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16963 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17409 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18422 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15623 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16830 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20193 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17104 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17921 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16838 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17235 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14397 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19205 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/16765 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14479 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15081 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21859 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15468 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15247 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19552 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15852 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13461 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15045 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4015 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19414 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15679 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
Looks good. Nice, thanks!

Since there is a high risk of breaking things with these python patches, please be especially cautious with the EWS and wait for almost all of them to finish before landing. If the API tests EWS becomes stuck in a loop (which happens if there are any problems with the autoinstaller) then please add the `skip-ews` label to break it, so we don't bother the EWS admins.

gsnedders committed:
Fix removed-in-Python-3.12 deprecations
https://bugs.webkit.org/show_bug.cgi?id=260878

Reviewed by Michael Catanzaro.

This largely just deals with the long-deprecated unittest deprecations:

 * assertRegexpMatches -> assertRegex
 * assertNotRegexpMatches -> assertNotRegex
 * assertRaisesRegexp -> assertRaisesRegex
 * assertEquals -> assertEqual

Additionally, this gets rid of usage of distutils (gone in Python
3.12) and imp (also gone in Python 3.12).

* Source/WebKit/Scripts/webkit/parser_unittest.py:
(UnsupportedPrecompilerDirectiveTest.assertRaisesRegex):
(UnsupportedPrecompilerDirectiveTest.test_error_at_else):
(UnsupportedPrecompilerDirectiveTest.test_error_at_elif):
* Tools/Scripts/webkitpy/benchmark_runner/benchmark_results_unittest.py:
(BenchmarkResultsTest.assertRaisesRegex):
(BenchmarkResultsTest.test_init):
* Tools/Scripts/webkitpy/benchmark_runner/http_server_driver/__init__.py:
* Tools/Scripts/webkitpy/benchmark_runner/utils.py:
(load_source):
(load_subclasses):
* Tools/Scripts/webkitpy/common/checkout/scm/scm_unittest.py:
(SCMTest.assertRegex):
(SCMTest):
(SCMTest.assertNotRegex):
(SCMTest._shared_test_diff_for_revision):
(test_commit_logs):
(GitTest.test_create_patch):
(GitTest.test_create_patch_with_git_index):
(GitTest.test_rename_files):
(GitSVNTest.test_delete_branch):
(GitSVNTest.test_remote_merge_base):
(GitSVNTest.test_commit_with_message_working_copy_only):
(GitSVNTest.test_commit_with_message):
(GitSVNTest.test_commit_with_message_git_commit):
(GitSVNTest.test_commit_with_message_git_commit_range):
(GitSVNTest.test_commit_with_message_only_local_commit):
(GitSVNTest.test_commit_with_message_multiple_local_commits_and_working_copy):
(GitSVNTest.test_commit_with_message_multiple_local_commits_always_squash):
(GitSVNTest.test_commit_with_message_multiple_local_commits):
(GitSVNTest.test_commit_with_message_not_synced):
(GitSVNTest.test_create_patch_local_plus_working_copy):
(GitSVNTest.test_create_patch):
(GitSVNTest.test_create_patch_after_merge):
(GitSVNTest.test_create_patch_with_changed_files):
(GitSVNTest.test_create_patch_git_commit):
(GitSVNTest.test_create_patch_git_commit_range):
(GitSVNTest.test_create_patch_working_copy_only):
(GitSVNTest.test_create_patch_multiple_local_commits):
(GitSVNTest.test_create_patch_not_synced):
(GitSVNTest.test_create_binary_patch):
* Tools/Scripts/webkitpy/common/prettypatch_unittest.py:
(PrettyPatchTest.check_ruby):
* Tools/Scripts/webkitpy/common/watchlist/watchlistparser_unittest.py:
(WatchListParserTest):
(WatchListParserTest.setUp):
* Tools/Scripts/webkitpy/common/webkitunittest.py: Removed.
* Tools/Scripts/webkitpy/layout_tests/servers/web_platform_test_server_unittest.py:
* Tools/Scripts/webkitpy/port/xvfbdriver_unittest.py:
(XvfbDriverTest.assertRaisesRegex):
(XvfbDriverTest.test_xvfb_not_replying):
* Tools/Scripts/webkitpy/tool/commands/applywatchlistlocal_unittest.py:
(ApplyWatchListLocalTest.assertRaisesRegex):
(test_args_parsing_with_two_bugs):
* Tools/Scripts/webkitpy/tool/commands/commandtest.py:
(CommandsTest):
* Tools/Scripts/webkitpy/w3c/test_converter_unittest.py:
(verify_test_harness_paths):

Canonical link: https://commits.webkit.org/267496@main
